### PR TITLE
DEV: Add `registerCustomCategorySectionLinkLockIcon` plugin API

### PR DESCRIPTION
**Pull request by tgxworld**
New client side plugin API that allows plugins or themes to customize
the fontawesome 5 icon used to indicate that a category is locked/read
restricted.

### CODE REVIEW

File: docs/CHANGELOG-JAVASCRIPT-PLUGIN-API.md [hunk 1]
```diff
@@ -7,6 +7,13 @@ in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## UNRELEASED
```
Comment (tgxworld) [inline docs/CHANGELOG-JAVASCRIPT-PLUGIN-API.md:10]: Marking this as unreleased first as intend to do the version bump along with other sidebar APIs being introduced.

File: app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js [hunk 2]
```diff
@@ -30,7 +33,7 @@ acceptance("Sidebar - Plugin API", function (needs) {
   let linkDidInsert, linkDestroy, sectionDestroy;
 
   test("Multiple header actions and links", async function (assert) {
-    withPluginApi("1.3.0", (api) => {
```
Comment (nattsw) [inline app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js:33]: It would've been nicer to do this update in another PR.

**Review by nattsw — APPROVED** [review]

**Comment by tgxworld** [comment]
Thank you @nattsw
